### ISSUE TITLE
"# pyright:" does not play nice along with "# pylint:" or "# pragma:" 

### ISSUE THREAD

**Issue by fametrano** [microsoft/pyright#1270]
**Describe the bug**
`pyright: reportMissingImports=false` is not respected when on the same line with `pylint: disable=import-error` or `# pragma: no cover`

**To Reproduce**
With Python > 3.6

```python
from datetime import datetime, timezone

# python 3.6
if sys.version_info.minor == 6:  # pragma: no cover
    import backports.datetime_fromisoformat  # pyright: reportMissingImports=false  # pylint: disable=import-error

    backports.datetime_fromisoformat.MonkeyPatch.patch_fromisoformat()
```

And also:

```python
from datetime import datetime, timezone

# python 3.6
if sys.version_info.minor == 6:  # pyright: reportMissingImports=false  # pragma: no cover
    import backports.datetime_fromisoformat  # pylint: disable=import-error

    backports.datetime_fromisoformat.MonkeyPatch.patch_fromisoformat()
```

**Expected behavior**
pyright should not complain about `Import "backports.datetime_fromisoformat" could not be resolved`

**VS Code extension or command-line**
VS Code extension Pylance 2020.12.2

**Additional context**
The following is a workaround, but it is suboptimal 
```python
from datetime import datetime, timezone

# python 3.6
if sys.version_info.minor == 6:  # pyright: reportMissingImports=false
    import backports.datetime_fromisoformat  # pragma: no cover  # pylint: disable=import-error

    backports.datetime_fromisoformat.MonkeyPatch.patch_fromisoformat()  # pragma: no cover
```

**Comment by erictraut** [comment]
Comments of the form `# pyright: xxx` are not per-line. They apply to the entire file. They are typically placed at or toward the beginning of the file.

**Comment by fametrano** [comment]
For a single line, should I use `# type: ignore`?

**Comment by erictraut** [comment]
The `# type: ignore` would apply to a single line, but it also won't play well with other comments on the same line. My recommendation is that if you need to disable the `reportMissingImports` diagnostic rule that you do it at the file level.

**Comment by fametrano** [comment]
Actually the following is fine:

```python
from datetime import datetime, timezone

if sys.version_info.minor == 6:  # pragma: no cover
    import backports.datetime_fromisoformat  # type: ignore # pylint: disable=import-error

    backports.datetime_fromisoformat.MonkeyPatch.patch_fromisoformat()
```

Thank you for your help, keep up with the good work. Pyright helped me to fix quite a few "possibly unbound" errors
